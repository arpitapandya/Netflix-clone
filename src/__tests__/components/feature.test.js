import React from "react";
import { render } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature />", () => {
    it("renders the <Feature /> with populated data", () => {
        const { container, queryByText, getByText } = render(
            <Feature>
                <Feature.Title>Unlimited films, TV programms and more.</Feature.Title>
                <Feature.SubTitle>Watch anywhere, Cancel at anytime.</Feature.SubTitle>
            </Feature>
        );

        expect(getByText("Unlimited films, TV programms and more.")).toBeTruthy();
        expect(queryByText("Watch anywhere, Cancel at anytime.")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it("renders the <Feature /> with just a title", () => {
        const { container, queryByText, getByText } = render(
            <Feature>
                <Feature.Title>Unlimited films, TV programms and more.</Feature.Title>
            </Feature>
        );

        expect(getByText("Unlimited films, TV programms and more.")).toBeTruthy();
        expect(queryByText("Watch anywhere, Cancel at anytime.")).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    });

    it("renders the <Feature /> with just a subtitle", () => {
        const { container, queryByText, getByText } = render(
            <Feature>
                <Feature.SubTitle>Watch anywhere. Cancel at anytime.</Feature.SubTitle>
            </Feature>
        );

        expect(queryByText("Unlimited films, TV programms and more.")).toBeFalsy();
        expect(getByText("Watch anywhere. Cancel at anytime.")).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});