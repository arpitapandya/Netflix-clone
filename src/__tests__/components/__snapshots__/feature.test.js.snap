// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> with just a subtitle 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cSqBsR"
>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 dQnJQV"
  >
    Watch anywhere. Cancel at anytime.
  </h2>
</div>
`;

exports[`<Feature /> renders the <Feature /> with just a title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cSqBsR"
>
  <h1
    class="feature__Title-sc-6swrp6-1 jiOEhx"
  >
    Unlimited films, TV programms and more.
  </h1>
</div>
`;

exports[`<Feature /> renders the <Feature /> with populated data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cSqBsR"
>
  <h1
    class="feature__Title-sc-6swrp6-1 jiOEhx"
  >
    Unlimited films, TV programms and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 dQnJQV"
  >
    Watch anywhere, Cancel at anytime.
  </h2>
</div>
`;
